packages/venia-ui/lib/util/apolloCache.js

```diff
@@ -63,3 +65,223 @@ export const cacheKeyFromType = object => {
             return defaultDataIdFromObject(object);
     }
 };
+
+const productKeyFieldFunction = object => {
+    return object.url_key
+        ? `${MagentoGraphQLTypes.ProductInterface}:${object.url_key}`
+        : defaultDataIdFromObject(object);
+};
+
+/**
+ * Replaces the deprecated cacheKeyFromType.
+ */
+export const TYPE_POLICIES = {
+    Query: {
+        fields: {
+            cart: {
+                // Replaces @connection(key: "Cart")
+                keyArgs: () => MagentoGraphQLTypes.Cart
+            }
+        }
+    },
+    Breadcrumb: {
+        // Specifying this key allows ApolloClient to use the cache instead of
+        // always fetching.
+        keyFields: ['category_id']
+    },
+    Cart: {
+        keyFields: () => MagentoGraphQLTypes.Cart,
+        fields: {
+            /**
+             * @client fields must be defined if queried along server props or
+             * the query will return null. See summary.gql.js
+             */
+            paymentNonce: {
+                read(cached) {
+                    return cached || null;
+                }
+            },
+            isBillingAddressSame: {
+                read(cached) {
+                    if (cached === false) {
+                        return cached;
+                    } else {
+                        return true;
+                    }
+                }
+            },
+            /*****/
+            applied_coupons: {
+                // eslint-disable-next-line no-unused-vars
+                merge(existing = [], incoming) {
```

Comment:
Yes, we are intentionally ignoring existing data. The `merge` function is used to specify how AC should deal with 
 incoming values of non-normalizable structure (an array or an object without an `id`).  In these cases where I have ignored `existing` there is an assumption that the `incoming` data will always reflect the true state of the data since it is "most recent". We also don't want to duplicate array values, which is what would happen if we concatenated into `existing`.

Now unfortunately, while my explanation applies to places we have used this style of `merge`, it seems like `applied_coupons` doesn't need a merge function since you can only apply a single one, and when you have no coupon applied the server returns `null` rather than an empty array. So I can remove this function but your concerns still apply to the other functions such as `items` or `available_payment_methods`.

Reply:
Okay, that clarifies it for me. Honestly, `no-unused-vars` should probably be [configured differently](https://eslint.org/docs/rules/no-unused-vars#args) anyway.